### ISSUE TITLE
Phalcon\Mvc\Model - getChangedFields() after getting related model.

### ISSUE THREAD

**Issue by naNuke** [phalcon/cphalcon#2257]
Phalcon: 1.3.1
MySQL: 5.5.35

So I have an issue here with checking changed fields.

This works:
```php
// Find post model.
$post = Post::findFirst(1);
// Set new title.
$post->title = 'new title';
// Did something change?
var_dump($post->getChangedFields());
```
This does not work:
(poster in this case in an alias of belongsTo relationship with user model)
```php
// Find post model.
$post = Post::findFirst(1);
// Get author's model.
$poster = $post->poster; // This causes the exception.
// Set new title.
$post->title = 'new title';
// Did something change?
var_dump($post->getChangedFields());
```
In the example above the model throws "*Change checking cannot be performed because the object has not been persisted or is deleted*"

Changing the example to use magic method instead of magic property makes it work again without a problem:
```php
// Find post model.
$post = Post::findFirst(1);
// Get author's model.
$poster = $post->getPoster(); // Magic method call doesn't cause exception anymore.
// Set new title.
$post->title = 'new title';
// Did something change?
var_dump($post->getChangedFields());
```

Not sure if this is intended behaviour or a bug but it is certainly weird :)

**Comment by andresgutierrez** [comment]
Fixed in 2.0.x

**Comment by nsossonko** [comment]
This is broken again somehow, see #11042
